apollo-federation/src/query_plan/fetch_dependency_graph.rs

```diff
@@ -616,17 +612,13 @@ impl FetchDependencyGraphNodePath {
                     .schema
                     .possible_runtime_types(typ)?
                     .into_iter()
-                    .map(|ctdp| ctdp.type_name)
```

Comment:
Refactors:
- We don't need to collect into intermediate vectors here, we can just flatten an iterator of iterators.